### ISSUE TITLE
BackendTrafficPolicy openapi schemas do not validate correctly

### ISSUE THREAD

**Issue by nicks** [envoyproxy/gateway#4746]
*Description*:
I'm trying to use kubeconform to validate BackendTrafficPolicy

I get this error:

```
stdin - BackendTrafficPolicy repos-new is invalid: problem validating schema. Check JSON formatting: jsonschema: '/spec/healthCheck/active/interval' does not validate with https://raw.githubusercontent.com/datreeio/CRDs-catalog/main/gateway.envoyproxy.io/backendtrafficpolicy_v1alpha1.json#/properties/spec/properties/healthCheck/properties/active/properties/interval/format: '3s' is not valid 'duration'
stdin - BackendTrafficPolicy admin-insights is invalid: problem validating schema. Check JSON formatting: jsonschema: '/spec/healthCheck/active/timeout' does not validate with https://raw.githubusercontent.com/datreeio/CRDs-catalog/main/gateway.envoyproxy.io/backendtrafficpolicy_v1alpha1.json#/properties/spec/properties/healthCheck/properties/active/properties/timeout/format: '20ms' is not valid 'duration'
```

*Additional info*:
I think the openapi schemas that we're generating are wrong. Particularly this line:

https://github.com/envoyproxy/gateway/blob/6c6633c2a8bfc772415ea3bf68e14ef20064cb22/charts/gateway-helm/crds/generated/gateway.envoyproxy.io_backendtrafficpolicies.yaml#L314

They say that `timeout` and `interval` are openapi durations. But they're not, they're a Kubernetes-specific duration format.

**Comment by arkodg** [comment]
thanks for raising this, do we need to edit these to `string` instead ?
The CEL validation on the `metav1.Duration` should still kick in and allow acceptable values
https://github.com/envoyproxy/gateway/blob/6c6633c2a8bfc772415ea3bf68e14ef20064cb22/api/v1alpha1/healthcheck_types.go#L81

**Comment by nicks** [comment]
ya, i poked around at how Cluster API is doing it, and i think they just leave the `kubebuilder` validation off entirely?
https://github.com/kubernetes-sigs/cluster-api/blob/781d1e4c39286d3d1e93c71557b5d02c2e78961b/api/v1beta1/machinehealthcheck_types.go#L101

and it will do the right thing with the existing `metav1.Duration` type?

**Comment by arkodg** [comment]
thanks for helping us find this @nicks 

we'll also need to add an entry in https://github.com/envoyproxy/gateway/blob/6c6633c2a8bfc772415ea3bf68e14ef20064cb22/tools/linter/golangci-lint/.golangci.yml#L34 to make sure this doesnt happen again

**Comment by Ninir** [comment]
Also faced this today for `retry.perRetry.timeout` on BackendTrafficPolicy. The duration having to be RFC3339-compliant (i.e PT60S) is not interpreted by the underlying engine, so kubeconform fails.

@arkodg As a newcomer to the Envoy Gateway world, is that something we can fix as a community? I'd love to give this a shot!

**Comment by arkodg** [comment]
@Ninir there's WIP PR https://github.com/envoyproxy/gateway/pull/4931 that I dropped the ball on, will pick up and make sure it gets into v1.4

**Comment by arkodg** [comment]
hey @Ninir would you interested in taking #4931 forward ?

**Comment by Ninir** [comment]
Sure! Is it about finishing the current work as in going through the last batch of files, or is there something else that would be required?
Happy to chat about it and make it cross the finish line! 🥰

**Comment by arkodg** [comment]
yeah @Ninir, its around making sure all API fields use `gwapiv1.Duration` instead of `metav1.Duration` . This will need to impacting three layers - `gateway-api` which reads the API field, the IR which is an intermediate representation and the `ids-translator` which set the IR values in the xds resources, tia 🙏
